### PR TITLE
Enhanced version information and Docker-like version command

### PR DESCRIPTION
**Pull request by funkyfuture**
I changed the version output, so also information about Python and docker-py are shown. Also implemented a command version to mimic Docker.

This is intended for trouble-shooting situations and to match Docker's behaviour.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Build is failing (also see a fatal error for a missing `yaml.h`)

**Comment by funkyfuture** [comment]
@thatJeztah, yeah, i saw that. but i have no idea how that may be caused by my commits. any advice how to deal with it?

**Comment by thaJeztah** [comment]
Don't know, maybe the build should just be restarted @aanand?

**Comment by funkyfuture** [comment]
to make clarify: 4117aaa doesn't mark a `--commit`-option as deprecated, since there is none. i meant the `--version` option. :-/

**Comment by funkyfuture** [comment]
built successfully after rebase.

**Comment by funkyfuture** [comment]
janky fails after rebase, unrelated to the changes themself.

**Comment by thaJeztah** [comment]
Is this the error it failed on?

    Running lint checks
    compose/cli/main.py:13:1: F401 '__version__' imported but unused

**Comment by thaJeztah** [comment]
So I think it is related; It's probably not an error, but it doesn't pass the lint checks. Looks like the import  on line 13 of `main.py` has to be removed to fix this

**Comment by funkyfuture** [comment]
ah, just looked at the last lines. thanks for the spot. will fix.

**Comment by funkyfuture** [comment]
hm, there are still errors ocuring. however i can't find any relation to my changes.

btw, this is how it looks like:
```
Docker-Compose version: 1.3.0dev
docker-py version: 1.2.2
CPython version: 2.7.3
```

File: compose/cli/main.py [hunk 1]
```diff
@@ -100,11 +98,13 @@ class TopLevelCommand(Command):
...
       up                 Create and start containers
       migrate-to-labels  Recreate containers to add labels
+      version            Show the Docker-Compose version information
 
     """
     def docopt_options(self):
         options = super(TopLevelCommand, self).docopt_options()
-        options['version'] = "docker-compose %s" % __version__
+        options['version'] = get_versions() \
+            + '\n\nThe Usage of --version is deprecated, use the command `version` instead.'
```
Comment (dnephin) [inline compose/cli/main.py]: I don't think this needs to be deprecated. `docker` supports both. The `--version` is just a short one-line version. I would leave this unchanged.

**Comment by dnephin** [comment]
Needs a rebase as well

**Comment by funkyfuture** [comment]
changedand rebased, here's the results;
```
root@22da0cb76426:/code# docker -v
Docker version 1.6.0, build 4749651

root@22da0cb76426:/code# docker-compose -v                                                                                                                                                                                                                              
Docker-Compose version: 1.3.0dev

root@22da0cb76426:/code# docker version
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): linux/amd64
Server version: 1.6.2
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 7c8fca2
OS/Arch (server): linux/amd64

root@22da0cb76426:/code# docker-compose version                                                                                                                                                                                                                         
Docker-Compose version: 1.3.0dev
docker-py version: 1.2.2
CPython version: 2.7.3

root@22da0cb76426:/code# docker-compose version --short
1.3.0dev
```

**Comment by dnephin** [comment]
LGTM

**Comment by funkyfuture** [comment]
`test_kill_signal_sigint` is failing… :roller_coaster:

**Comment by dnephin** [comment]
I'll see what I can do about those tests

File: compose/cli/utils.py [hunk 2]
```diff
@@ -120,3 +123,15 @@ def is_mac():
 
 def is_ubuntu():
     return platform.system() == 'Linux' and platform.linux_distribution()[0] == 'Ubuntu'
+
+
+def get_version_info(scope):
+    versioninfo = 'Docker-Compose version: %s' % __version__
```
Comment (aanand) [inline compose/cli/utils.py]: This should either say `docker-compose` or `Docker Compose`. We're not consistent across the platform (Machine uses the first form, Engine the second) but I vote for the first.

**Comment by funkyfuture** [comment]
thanks for the spot. fixed.

**Comment by dnephin** [comment]
Looks like tests are passing.

LGTM

**Comment by aanand** [comment]
LGTM

**Pull request merged**